operators/pkg/instctrl/containers.go

```diff
@@ -89,6 +110,22 @@ func (r *InstanceReconciler) enforceContainer(ctx context.Context) error {
 		// either rejected or cause the restart of the Pod, with consequent possible data loss.
 		if depl.CreationTimestamp.IsZero() {
 			depl.Spec = forge.DeploymentSpec(instance, environment, &r.ContainerEnvOpts)
+
+			if environment.MountPathEnabled {
+				volume := v1.Volume{Name: NFSVolumeName}
+				volumeMount := v1.VolumeMount{Name: NFSVolumeName}
+
+				if err := r.GetNFSVolume(ctx, &volume, &volumeMount); err != nil {
+					return err
+				}
+
+				depl.Spec.Template.Spec.Volumes = append(depl.Spec.Template.Spec.Volumes, volume)
+				for i := range depl.Spec.Template.Spec.Containers {
+					if depl.Spec.Template.Spec.Containers[i].Name == environment.Name {
+						depl.Spec.Template.Spec.Containers[i].VolumeMounts = append(depl.Spec.Template.Spec.Containers[i].VolumeMounts, volumeMount)
+					}
+				}
+			}
```

Comment:
Kubernetes objects forging logic shall be in forge, and not here. That logic shall not include calls to the Kubernetes APIs (i.e., it shall never fail), hence the retrieval of the NFS URL and path shall be performed in this method in advance.

Reply:
Thinking of a way of moving it

Reply:
A possible solution has been provided, take a look at it. I don't know if the approach is completely correct.